roles/bigboot/files/bigboot.sh

```diff
@@ -263,30 +261,24 @@ deactivate_volume_group() {
 
 check_available_free_space() {
     local device="${DEVICE_NAME}${PARTITION_PREFIX}${ADJACENT_PARTITION_NUMBER}"
-    free_device_space_in_bytes=$(get_free_device_size)
-    # if there is enough free space after the adjacent partition, there is no need to shrink it.
-    if [[ $free_device_space_in_bytes -gt $INCREMENT_BOOT_PARTITION_SIZE_IN_BYTES ]]; then
-        SHRINK_SIZE_IN_BYTES=0
-        return
+    # Get LVM details
+    eval "$(/usr/sbin/lvm pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count $device)"
```

Comment:
I'm probably missing something here. What are you getting out of this call?

Reply:
It sets all the LVM info variables we need with one command.

Reply:
IIUC this command displays information about the PV, but the information is not saved or used anywhere afterwards. Is it?

Reply:
> IIUC this command displays information about the PV, but the information is not saved or used anywhere afterwards. Is it?

It is. The output of the `pvs` command would be something like this example:

```
# pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count /dev/nvme0n1p2 
  LVM2_VG_NAME='VolGroup00' LVM2_VG_EXTENT_SIZE='4194304' LVM2_PV_PE_COUNT='8072' LVM2_PV_PE_ALLOC_COUNT='5120' LVM2_VG_FREE_COUNT='2952'
```

By using the command substitution and `eval` builtin, bash sets variables using the names and values output by the `pvs` command. Here's an example to demonstrate some of the vars being set:

```
# device=/dev/nvme0n1p2
# eval "$(/usr/sbin/lvm pvs --noheadings --nameprefixes --nosuffix --units b -o vg_name,vg_extent_size,pv_pe_count,pv_pe_alloc_count,vg_free_count $device)"
# echo $LVM2_VG_NAME
VolGroup00
# echo $LVM2_VG_EXTENT_SIZE
4194304
```

Reply:
Interesting. I didn't know that. Thanks for the explanation